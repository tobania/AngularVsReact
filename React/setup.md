# React Setup

## Prerequirements
- nodejs
- Backend runs in the background and is accessable

# Setup
Navigate to the frontend root (src) and execute the following commands:
npm install
npm start
