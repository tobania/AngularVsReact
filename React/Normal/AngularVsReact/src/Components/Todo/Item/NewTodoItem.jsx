import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import AddTodoItem from '../../../HttpServices/TodoItems/AddTodoItemService';

class Todo extends PureComponent {
  constructor() {
    super();
    this.state = { text: '' };

    this.handleChange = this.handleChange.bind(this);
    this.addTodoItem = this.addTodoItem.bind(this);
  }

  handleChange(event) {
    this.setState({ text: event.target.value });
  }

  addTodoItem() {
    const { text } = this.state;
    const { addCreatedTodoItem, identifier } = this.props;
    AddTodoItem(identifier, { text }, addCreatedTodoItem);

    this.setState({ text: '' });
  }

  render() {
    const { text } = this.state;
    return (
      <div className="row mt-4">
        <div className="col-10">
          <input type="text" className="form-control" value={text} onChange={this.handleChange} />
        </div>
        <div className="col-2">
          <button type="button" className="btn btn-success float-right" onClick={this.addTodoItem}>+</button>
        </div>
      </div>
    );
  }
}

Todo.propTypes = {
  identifier: PropTypes.string.isRequired,
  addCreatedTodoItem: PropTypes.func.isRequired,
};

Todo.defaultProps = {
};

export default Todo;
