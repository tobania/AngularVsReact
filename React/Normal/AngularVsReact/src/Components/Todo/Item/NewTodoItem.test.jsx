import React from 'react';
import renderer from 'react-test-renderer';
import NewTodoItem from './NewTodoItem';

describe('NewTodoItem', () => {
  it('gets loaded correctly', () => {
    const component = renderer.create(
      <NewTodoItem identifier="identifier" addCreatedTodoItem={jest.fn} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handle change should update the state', () => {
    const component = renderer.create(
      <NewTodoItem identifier="identifier" addCreatedTodoItem={jest.fn} />,
    ).getInstance();

    expect(component.state.text).toBe('');
    component.handleChange({ target: { value: 'test' } });
    expect(component.state.text).toBe('test');
  });

  it('addTodoItem should update the state', () => {
    const component = renderer.create(
      <NewTodoItem identifier="identifier" addCreatedTodoItem={jest.fn} />,
    ).getInstance();

    component.setState({ text: 'test' });
    expect(component.state.text).toBe('test');
    component.addTodoItem();
    expect(component.state.text).toBe('');
  });
});
