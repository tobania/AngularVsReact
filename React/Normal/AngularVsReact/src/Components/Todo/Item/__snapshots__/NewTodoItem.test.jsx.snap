// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoItem gets loaded correctly 1`] = `
<div
  className="row mt-4"
>
  <div
    className="col-10"
  >
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="col-2"
  >
    <button
      className="btn btn-success float-right"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
</div>
`;
